src/renderer/services/maps-manager.service.ts

```diff
@@ -162,6 +213,17 @@ export class MapsManagerService {
         return lastValueFrom(this.ipcService.sendV2("unregister-maps-deep-link"));
     }
 
+    public addImportListener(listener: (map: BsmLocalMap, version?: BSVersion) => void): void {
+        this.importListeners.push(listener);
+    }
+
+    public removeImportListener(listener: (map: BsmLocalMap, version?: BSVersion) => void): void {
+        const index = this.importListeners.indexOf(listener);
```

Comment:
I wonder what happen if the same callback is added multiple time 🤔 \

Like:
```ts
const cb = () => {};
addImportListener(cb);
addImportListener(cb);
addImportListener(cb);

removeImportListener(cb);
```
Maybe a `Set` of callbacks would work better here instead of an array 🤔
I let you test that 😉

Reply:
Yeah might be better to do a `Set` in any case, but we'll see if that works.

---

Actually there was a bug is this code as well, where the listeners weren't removed properly. Not sure if the fix should be here since the PR might take some more time to get merged, but the snippet bugged code is here.

In master currently
`local-maps-list-panel.component.tsx`
```js
    useEffect(() => {
        if (isActiveOnce) {
            loadMaps();
        }

        return () => {
            setMaps(null);
            loadPercent$.next(0);
            subs.forEach(s => s.unsubscribe());
            mapsDownloader.removeOnMapDownloadedListener(loadMaps); // should not be loadMaps
        };
    }, [isActiveOnce, version, linked]);

    useEffect(() => {
        if(isActiveOnce){
            // The listener that was being added multiple times
            mapsDownloader.addOnMapDownloadedListener((map, targetVersion) => {
                if (!equal(targetVersion, version)) {
                    return;
                }
                setMaps((maps ? [map, ...maps] : [map]));
            });
        }

        return () => {
            mapsDownloader.removeOnMapDownloadedListener(loadMaps); // should not be loadMaps
        }

    }, [isActiveOnce, version, maps])
```

Reply:
👀 Yeah I will fix that in on master 😅